### PR TITLE
Add VBAS and VMAX signals to PCDSMotorBase

### LINKED ISSUE #1224: Base Velocity and Velocity Max missing from EpicsMotor base class in pcdsdevices.epics_motor
## Current Behavior
`<PV>:m<N>.VMAX` and `<PV>:m<N>.VBAS` are not ported over in `EpicsMotor` parent class. This means that if either of those PVs get set (like in the `edm` screens, for example...) then you run into unclear limitations to changing the velocities in inherited screens. This is particularly troublesome for SmarActs, as you become unable to uncap the velocities, or may even be inexplicably stuck at certain velocities if the `VBAS` was set erroneously.

## Expected Behavior
`VMAX` and `VBAS` PVs from `EpicsMotor` parent class available in a `kind=config` type signal in Typhos screens.

## Context / environment
After changing over the out-of-loop mechanical delay lines in the FTL to the typhos screens, users (myself included) were unable to change the velocities on several stages with no clear indicator as to why.

## Steps to Reproduce (for bugs)
1. `caput` the `VBAS` or `VMAX` to some nominal value (e.g. 1 mm/s)
2. Try to change the `velocity` signal in a `pcdsdevices.epics_motor.SmarAct` screen that is below either of those values

## Suggested Solution
Bring these two signals over in the `tab_whitelist` in the parent class to prevent this issue in the future.

### PR DESCRIPTION
**Pull request by slactjohnson**
## Description
Moves VBAS and VMAX signals to PCDSMotorBase for access by more sub-classes. 
Change SmarAct to use PCDSMotorBase. 

## Motivation and Context
Closes #1224 

## How Has This Been Tested?
Tested using the same test motor @aberges-SLAC had in his submitted issue. 

## Where Has This Been Documented?
This PR.

## Screenshots:
Typhos suite of that motor with "omitted" signals also shown. 

image

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [x] Test suite passes on GitHub Actions
- [x] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

File: pcdsdevices/epics_motor.py [hunk 1]
```diff
@@ -1470,7 +1473,7 @@ def done_comparator(self, readback, setpoint):
         return setpoint-self.atol < readback < setpoint+self.atol
 
 
-class SmarAct(EpicsMotorInterface):
+class SmarAct(PCDSMotorBase):
```
Comment (ZLLentz) [inline pcdsdevices/epics_motor.py:1476]: Does the SmarAct class have the same oddities that the PCDSMotorBase was created to fix?
```
        1. The 'TDIR' field does not exist on the PCDS implementation. This
        indicates what direction the motor has travelled last. To account for
        this difference, we use the :meth:`._pos_changed` callback to keep
        track of which direction we believe the motor to be travelling
        and store this in a simple :class:`~ophyd.signal.Signal`.
        2. The 'SPG' field implements the three states used in the LCLS
        motor record.  This is a reduced version of the standard EPICS
        'SPMG' field.  Setting to 'STOP', 'PAUSE' and 'GO'  will respectively
        stop motor movement, pause a move in progress, or resume a paused move.
        3. In some cases, puts to the setpoint PV cannot be waited on. In fact
        this seems to brick the whole python session, requiring a slightly
        modified set_current_position method.
```
If not, we should instead be moving these all the way up to `EpicsMotorInterface`.

Comment (slactjohnson) [reply pcdsdevices/epics_motor.py:1476]: Ah, you're right. It does not. I'll move them.

**Comment by aberges-SLAC** [comment]
If you want to be my hero, you could also fix line 1516 for me and close [issue #1235](https://github.com/pcdshub/pcdsdevices/issues/1235) :]]

**Comment by slactjohnson** [comment]
@aberges-SLAC I could do that, but we generally don't do things that way. Usually we're just fixing one or more related bugs, in one PR. That particular bug is not related to the original intent of the PR. But I get it, it's just two letters. 

@ZLLentz What do you think about bundling Adam's requested fix in a separate PR vs. this PR? Not just with this PR, but rather what is our general stance on that kind of thing?

**Comment by aberges-SLAC** [comment]
> @aberges-SLAC I could do that, but we generally don't do things that way. Usually we're just fixing one or more related bugs, in one PR. That particular bug is not related to the original intent of the PR. But I get it, it's just two letters.
> 
> @ZLLentz What do you think about bundling Adam's requested fix in a separate PR vs. this PR? Not just with this PR, but rather what is our general stance on that kind of thing?

It's most definitely bad practice. :] I should really just do the right thing and open a separate PR against the issue and have it tracked, just in case.

**Comment by ZLLentz** [comment]
In general, the code is much easier to review and discuss in a focused way if it only addresses a limited number of related things. Bundling should be avoided when possible.

I wouldn't reject a PR for bundling unless it was excessive (tons of lines changed with multiple unrelated goals)

**Comment by slactjohnson** [comment]
@ZLLentz Sounds good, thanks for confirming my suspicions.
@aberges-SLAC I will finish up this PR and make a separate one to fix the tip-tilt class.

**Review by tangkong — APPROVED** [review]
This looks good to me 👍

**Pull request merged**